packages/deployments/contracts/contracts/facets/StableSwapFacet.sol

```diff
@@ -0,0 +1,458 @@
+   * @param tokenIndex index of which token will be withdrawn
+   * @return availableTokenAmount calculated amount of underlying token
+   * available to withdraw
+   */
+  function calculateRemoveLiquidityOneToken(
+    bytes32 canonicalId,
+    uint256 tokenAmount,
+    uint8 tokenIndex
+  ) external view returns (uint256 availableTokenAmount) {
+    return s.swapStorages[canonicalId].calculateWithdrawOneToken(tokenAmount, tokenIndex);
+  }
+
+  /**
+   * @notice This function reads the accumulated amount of admin fees of the token with given index
+   * @param canonicalId the canonical token id
+   * @param index Index of the pooled token
+   * @return admin's token balance in the token's precision
+   */
+  function getAdminBalance(bytes32 canonicalId, uint256 index) external view returns (uint256) {
+    return s.swapStorages[canonicalId].getAdminBalance(index);
+  }
+
+  /*** STATE MODIFYING FUNCTIONS ***/
+
+  /**
+   * @notice Swap two tokens using this pool
+   * @param canonicalId the canonical token id
+   * @param tokenIndexFrom the token the user wants to swap from
+   * @param tokenIndexTo the token the user wants to swap to
+   * @param dx the amount of tokens the user wants to swap from
+   * @param minDy the min amount the user would like to receive, or revert.
+   * @param deadline latest timestamp to accept this transaction
+   */
+  function swap(
+    bytes32 canonicalId,
+    uint8 tokenIndexFrom,
+    uint8 tokenIndexTo,
+    uint256 dx,
+    uint256 minDy,
+    uint256 deadline
+  ) external nonReentrant deadlineCheck(deadline) returns (uint256) {
+    return s.swapStorages[canonicalId].swap(tokenIndexFrom, tokenIndexTo, dx, minDy);
+  }
+
+  /**
+   * @notice Swap two tokens using this pool
+   * @param canonicalId the canonical token id
+   * @param assetIn the token the user wants to swap from
+   * @param assetOut the token the user wants to swap to
+   * @param amountIn the amount of tokens the user wants to swap from
+   */
+  function swapExact(
+    bytes32 canonicalId,
+    uint256 amountIn,
+    address assetIn,
+    address assetOut,
+    uint256 minAmountOut,
+    uint256 deadline
+  ) external payable nonReentrant deadlineCheck(deadline) returns (uint256) {
+    uint8 tokenIndexFrom = getTokenIndex(canonicalId, assetIn);
+    uint8 tokenIndexTo = getTokenIndex(canonicalId, assetOut);
+    return s.swapStorages[canonicalId].swap(tokenIndexFrom, tokenIndexTo, amountIn, minAmountOut);
+  }
+
+  /**
+   * @notice Add liquidity to the pool with the given amounts of tokens
+   * @param canonicalId the canonical token id
+   * @param amounts the amounts of each token to add, in their native precision
+   * @param minToMint the minimum LP tokens adding this amount of liquidity
+   * should mint, otherwise revert. Handy for front-running mitigation
+   * @param deadline latest timestamp to accept this transaction
+   * @return amount of LP token user minted and received
+   */
+  function addStableLiquidity(
```

Comment:
I feel like even though it's not an ERC standard, we should conform with the same method names as Curve just to make this recognizable.

Like instead of `addStableLiquidity` just stick to `addLiquidity`

Besides, it's not like the function name is going to enforce that the liquidity is stablecoins lol

Reply:
Yeah, i love that. `addLiquidity` but we have already such a function on `RoutersFacet`. so i had to think another name.
do you have any idea?

Reply:
ah, so if that's the issue, we should probs just call it `addStableSwapLiquidity` lol

Reply:
if there's really an overlap... maybe we should be calling `RoutersFacet.addLiquidity` `RoutersFacet.addRouterLiquidity` lol

but dwbi for rn